scripts/create_eks_cluster.sh

```diff
@@ -0,0 +1,369 @@
+REGION=					# cluster region (MANDATORY)
+NODEGROUP_NAME=$CLUSTER_NAME-nodegroup	# managed nodegroup name
+LT_NAME=$CLUSTER_NAME-lt		# EC2 launch template name
+KEYNAME=				# keypair name for ssh access to worker nodes
+SSH_SECURITY_GROUP_NAME="$CLUSTER_NAME-ssh-allow"
+SSH_ALLOW_CIDR="0.0.0.0/0"		# source IP from which ssh access is allowed
+INSTANCE_TYPE=m5.large			# EC2 instance type
+INSTANCE_NUM=2				# Number of instances in cluster
+## Calico/VPP deployment yaml; could be url or local file
+CALICO_VPP_YAML=https://raw.githubusercontent.com/projectcalico/vpp-dataplane/master/yaml/generated/calico-vpp-eks-dpdk.yaml
+#CALICO_VPP_YAML=<full path>/calico-vpp-eks-dpdk.yaml
+## init_eks.sh script location; could be url or local file
+INIT_EKS_SCRIPT=https://raw.githubusercontent.com/projectcalico/vpp-dataplane/master/scripts/init_eks.sh
+#INIT_EKS_SCRIPT=<full path>/init_eks.sh
+###############################################################################
+
+usage ()
+{
+	echo
+	echo "Either execute the script after filling in the CONFIG PARAMS section in the"
+	echo "script or execute the script with command-line options as follows:"
+	echo
+	echo "    bash $0  <cluster name>  -r <region name>  [-k <keyname>]"
+	echo "    [-t <instance type>]  [-n <number of instances>]  [-f <calico/vpp config yaml file>]"
+	echo
+	echo "Mandatory options are \"cluster name\" and \"region name\"."
+	exit 1
+}
+
+### Parse the command line
+if [ "$1" != "" ]; then
+	CLUSTER_NAME=$1
+	shift
+	while getopts ":r:k:t:n:f:" flag
+	do
+	        case "${flag}" in
+	                r) REGION=${OPTARG};;
+	                k) KEYNAME=${OPTARG};;
+	                t) INSTANCE_TYPE=${OPTARG};;
+	                n) INSTANCE_NUM=${OPTARG};;
+	                f) CALICO_VPP_YAML=${OPTARG};;
+	               \?) echo "Invalid option : -${OPTARG}"; usage;;
+	                :) echo "Option -${OPTARG} requires an argument"; usage;;
+	        esac
+	done
+	NODEGROUP_NAME=$CLUSTER_NAME-nodegroup
+	LT_NAME=$CLUSTER_NAME-lt
+	SSH_SECURITY_GROUP_NAME="$CLUSTER_NAME-ssh-allow"
+fi
+
+### Check for mandatory options
+if [ "$CLUSTER_NAME" = "" -o "$REGION" = "" ]; then
+	echo "Please provide \"cluster name\" and \"region name\"."
+	echo
+	usage
+	exit 1
+fi
+
+#### Basic sanity checks
+# Save pain of tons of unnecessary errors and wasted time since EKS cluster
+# creation takes an awfully long time (>20 mins)
+
+### check for OS type; only macos and linux
+OS_TYPE=`uname`
+echo; echo "OS_TYPE=$OS_TYPE"
+if [ "$OS_TYPE" = "Darwin" ]; then
+	BASE64_COMMAND="base64"
+elif [ "$OS_TYPE" = "Linux" ]; then
+	# On Linux, by default base64 wraps the lines to 76 column width
+	# which causes launch template creation to fail. "--wrap=0" disables it.
+	BASE64_COMMAND="base64 --wrap=0"
+else
+        echo
+        echo "ERROR: OS_TYPE=$OS_TYPE not supported"
+        exit 1
+fi
+echo "BASE64_COMMAND=$BASE64_COMMAND"
+
+
+### Make sure we have aws cli version 2
+which aws
```

Comment:
Could be nice to redirect the output of the three `which` commands to `/dev/null` to keep the script output a bit cleaner as well :)

Reply:
On the contrary, I want to see them :)